### ISSUE TITLE
[v2] Android run error "react-native-navigation:compileReactNative55DebugJavaWithJavac FAILED"

### ISSUE THREAD

**Issue by herarya** [wix/react-native-navigation#3674]
### Issue Description

[ENTER DESCRIPTION HERE - Please make sure to use the current [naming conventions](https://github.com/wix/react-native-navigation/issues/197)]

hi, i after run android error "compileReactNative55DebugJavaWithJavac FAILED" but ios work, what version for rn-` 0.55.3`?

### Steps to Reproduce / Code Snippets / Screenshots

[FILL THIS OUT - It will be extremely helpful]
`> Task :react-native-navigation:compileReactNative55DebugJavaWithJavac FAILED
/Users/Weefer/Documents/iTakejob/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/interfaces
/ScrollEventListener.java:5: error: cannot find symbol
import com.facebook.react.uimanager.events.EventDispatcherListener;
                                          ^
  symbol:   class EventDispatcherListener
  location: package com.facebook.react.uimanager.events
/Users/Weefer/Documents/iTakejob/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/interfaces/ScrollEventListener.java:10: error: cannot find symbol
public class ScrollEventListener implements EventDispatcherListener {
                                            ^
  symbol: class EventDispatcherListener
/Users/Weefer/Documents/iTakejob/node_modules/react-native-navigation/lib/android/app/src/main/java/com/reactnativenavigation/react/ReactView.java:12: error: cannot find symbol
import com.facebook.react.uimanager.JSTouchDispatcher;
....`

---
### Environment
* React Native Navigation version: `2.0.2397`
* React Native version: `0.55.3`
* Platform(s) (iOS, Android, or both?): ANDROID
* Device info (Simulator/Device? OS version? Debug/Release?): ANDROID NEXUS

**Comment by guyca** [comment]
Please follow the instructions in the [installation guide](https://wix.github.io/react-native-navigation/v2/#/docs/Installing?id=_2-updating-gradle-build-files) regarding build flavors

**Comment by herarya** [comment]
@guyca i'm already follow installation guide

**#setting gradle**
```
include ':react-native-navigation'
project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/lib/android/app/')
```
**build gradle**

`buildscript {
    repositories {
        google()
        mavenLocal()
        mavenCentral()
        jcenter()
        maven {
            url 'https://maven.google.com/'
            name 'Google'
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.3'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                variantFilter { variant ->
                    def names = variant.flavors*.name
                    // To check for a certain build type, use variant.buildType.name == "<buildType>"
                    if (names.contains("reactNative51") || names.contains("reactNative55")) {
                        // Gradle ignores any variants that satisfy the conditions above.
                        setIgnore(true)
                    }
                }
            }
        }
    }
}

allprojects {
    repositories {
        mavenLocal()
        mavenCentral()
        google()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../../node_modules/react-native/android"
        }
        maven { url 'https://jitpack.io' }
    }
}

ext {
    buildToolsVersion = "27.0.3"
    minSdkVersion = 19
    compileSdkVersion = 26
    targetSdkVersion = 25
    supportLibVersion = "27.0.3"
}
`

android/build gradle

`apply plugin: "com.android.application"

import com.android.build.OutputFile

/**
 * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
 * and bundleReleaseJsAndAssets).
 * These basically call `react-native bundle` with the correct arguments during the Android build
 * cycle. By default, bundleDebugJsAndAssets is skipped, as in debug/dev mode we prefer to load the
 * bundle directly from the development server. Below you can see all the possible configurations
 * and their defaults. If you decide to add a configuration block, make sure to add it before the
 * `apply from: "../../node_modules/react-native/react.gradle"` line.
 *
 * project.ext.react = [
 *   // the name of the generated asset file containing your JS bundle
 *   bundleAssetName: "index.android.bundle",
 *
 *   // the entry file for bundle generation
 *   entryFile: "index.android.js",
 *
 *   // whether to bundle JS and assets in debug mode
 *   bundleInDebug: false,
 *
 *   // whether to bundle JS and assets in release mode
 *   bundleInRelease: true,
 *
 *   // whether to bundle JS and assets in another build variant (if configured).
 *   // See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Build-Variants
 *   // The configuration property can be in the following formats
 *   //         'bundleIn${productFlavor}${buildType}'
 *   //         'bundleIn${buildType}'
 *   // bundleInFreeDebug: true,
 *   // bundleInPaidRelease: true,
 *   // bundleInBeta: true,
 *
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true,
 *   // The configuration property can be in the following formats
 *   //         'devDisabledIn${productFlavor}${buildType}'
 *   //         'devDisabledIn${buildType}'
 *
 *   // the root of your project, i.e. where "package.json" lives
 *   root: "../../",
 *
 *   // where to put the JS bundle asset in debug mode
 *   jsBundleDirDebug: "$buildDir/intermediates/assets/debug",
 *
 *   // where to put the JS bundle asset in release mode
 *   jsBundleDirRelease: "$buildDir/intermediates/assets/release",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in debug mode
 *   resourcesDirDebug: "$buildDir/intermediates/res/merged/debug",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in release mode
 *   resourcesDirRelease: "$buildDir/intermediates/res/merged/release",
 *
 *   // by default the gradle tasks are skipped if none of the JS files or assets change; this means
 *   // that we don't look at files in android/ or ios/ to determine whether the tasks are up to
 *   // date; if you have any other folders that you want to ignore for performance reasons (gradle
 *   // indexes the entire tree), add them here. Alternatively, if you have JS files in android/
 *   // for example, you might want to remove it from here.
 *   inputExcludes: ["android/**", "ios/**"],
 *
 *   // override which node gets called and with what additional arguments
 *   nodeExecutableAndArgs: ["node"],
 *
 *   // supply additional arguments to the packager
 *   extraPackagerArgs: []
 * ]
 */

project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion

    defaultConfig {
        applicationId "com.itakejob"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative56"
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

configurations.all {
    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
        def requested = details.requested
        if (requested.group == 'com.android.support') {
            details.useVersion "26.1.0"
        }
    }
}

dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation project(':react-native-navigation')
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}
`

**Comment by NicolasDionB** [comment]
@herarya Did you find a fix for this? I have the same issue, even if I followed the installation guide... I chose the 2nd method for the flavor, which is "Ignore other RNN flavors". I'm targeting version 55...

**Comment by herarya** [comment]
@NicolasDionB im using this version 
`react-native-navigation": "^2.0.2427"`
`"react-native": "0.56.0"`

**Comment by NicolasDionB** [comment]
@herarya Thanks but no luck, that's what I'm also using and still not working. I was using RN 55 but I tried to upgrade to 56 and got the same result. I was digging yesterday and it seemed to be some import issue with the react-native dependency. I got linkage errors in Android Studio and furthermore it seemed like the class definition for the react-native lib was using 0.20.1! That's the one I get with graddle (see [this](https://github.com/facebook/react-native/issues/12309) post).

I wasn't able to change this line: `implementation "com.facebook.react:react-native:+"` to something like this: `implementation "com.facebook.react:react-native:0.56.0"` which should be used in fact. I then tried to add this line in android/build.graddle: `maven { url "$rootDir/../node_modules/react-native/android" }` along with the line with the url provided in the installation guide (`'https://jitpack.io'`). Still no luck, I don't get import/linkage errors in Android Studio but I still get the same result upon building the app.

Ideas anyone?
